tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ServiceStackRedisTests.cs

```diff
@@ -26,68 +30,52 @@ public ServiceStackRedisTests(ITestOutputHelper output)
         [SkippableTheory]
         [MemberData(nameof(PackageVersions.ServiceStackRedis), MemberType = typeof(PackageVersions))]
         [Trait("Category", "EndToEnd")]
-        public void SubmitsTraces(string packageVersion)
+        public async Task SubmitsTraces(string packageVersion)
         {
             using var telemetry = this.ConfigureTelemetry();
             using (var agent = EnvironmentHelper.GetMockAgent())
             using (RunSampleAndWaitForExit(agent, arguments: $"{TestPrefix}", packageVersion: packageVersion))
             {
+#if NETCOREAPP3_1_OR_GREATER
+                var numberOfRuns = 3;
+#else
+                var numberOfRuns = 2;
+#endif
+
+                var expectedSpansPerRun = 9;
+
                 // note: ignore the INFO command because it's timing is unpredictable (on Linux?)
-                var spans = agent.WaitForSpans(11)
+                var spans = agent.WaitForSpans(numberOfRuns * expectedSpansPerRun)
                                  .Where(s => s.Type == "redis" && s.Resource != "INFO" && s.Resource != "ROLE" && s.Resource != "QUIT")
                                  .OrderBy(s => s.Start)
                                  .ToList();
+                spans.Count.Should().Be(numberOfRuns * expectedSpansPerRun);
```

Comment:
I just wanted to double check the number of spans we actually expect here, given we filter out `INFO`, `ROLE` and `QUIT`, and we've changed from expecting 11 to 18/27... 

I assume the 11 we used before was already far too small so maybe there's no issue, but if we "expect" fewer fans than we generate than we can end up missing spans that are sent later in the test (leading to flakiness)

Reply:
After some debugging I found that there is one of each of those commands per routine, so I'm including them in the WaitForSpans call in e45a6ace9a23adb8539eccae2762fceee3f12c6a. This means we should be waiting for the exact number of spans now, and should not miss spans at the end

Reply:
> I found that there is one of each of those commands per routine,

In which case, that begs the quest of why are we ignoring them I guess 🤔 Given we can get a stable ordering with `raw_command`, seems like we should just include them?

Also, we won't _necessarily_ find out quickly if one of these is missing (for example) - `WaitForSpans()` doesn't throw if it doesn't get the expected number of spans, so it would just make the test run for a _loooong_ time, which will be a bit annoying (as we probably won't notice)

Reply:
> Given we can get a stable ordering with raw_command, seems like we should just include them

Very good point. I've added it in 5ef25346593216714cea5b29613a0474f844c904 . If for some reason it becomes unstable we can adjust 👍🏼